### PR TITLE
JSdocs for autocomplete, lookup, and search components

### PR DESCRIPTION
**Pull request by Blackbaud-DeniseCisneros**

### CODE REVIEW

**Comment by blackbaud-ado** [comment]
Preview build at [https://host.nxt.blackbaud.com/skyux-lookup?_sp.spa=pr-search-lookup-autocomplete-jsdocs](https://host.nxt.blackbaud.com/skyux-lookup?_sp.spa=pr-search-lookup-autocomplete-jsdocs)

File: src/app/public/modules/autocomplete/autocomplete.component.ts [hunk 1]
```diff
@@ -137,15 +173,29 @@ export class SkyAutocompleteComponent
...
   public searchResultsLimit: number;
 
   @Input()
   public noResultsFoundText: string;
 
+/**
+ * Fires when users select items in the dropdown list.
+ */
   @Output()
   public get selectionChange(): EventEmitter<SkyAutocompleteSelectionChange> {
```
Comment (Blackbaud-AlexKingman) [inline src/app/public/modules/autocomplete/autocomplete.component.ts:200]: This event is showing as "Optional." That doesn't really make sense for an event, and is probably a docs-tools bug. I filed an issue here: https://github.com/blackbaud/skyux-docs-tools/issues/132

**Review by Blackbaud-AlexKingman — COMMENTED** [review]
Left some comments for the autocomplete component. Also we'll need to...
1. Add `@internal` to the following: `SkyAutocompleteAdapterService`, `SkyAutocompleteDefaultSearchFunctionOptions`, `SkyAutocompleteInputTextChange`, and `SkyAutocompleteSearchResult`.
2. In the `SkyAutocompleteInputDirective`, change the @default value for disabled to `false` (need to remove the quotes b/c its a boolean not a string).

File: src/app/public/modules/autocomplete/autocomplete.component.ts [hunk 2]
```diff
@@ -105,6 +124,12 @@ export class SkyAutocompleteComponent
...
   }
 
+/**
+ * Specifies a function to dynamically manage the data source when users
+ * change the text in the autocomplete field. The search function must return
+ * an array or a promise of an array. The `search` property is particularly
+ * useful when the data source does not live in the source code.
+ */
   @Input()
   public set search(value: SkyAutocompleteSearchFunction) {
```
[the remark this answers is not in the document]
Comment (Blackbaud-AlexKingman) [reply src/app/public/modules/autocomplete/autocomplete.component.ts:134]: Yep, its a bug. Filed: https://github.com/blackbaud/skyux-docs-tools/issues/133
This shouldn't hold up your PR though.

File: src/app/public/modules/lookup/lookup-autocomplete-adapter.ts [hunk 3]
```diff
@@ -12,30 +12,79 @@ import {
...
   public propertiesToSearch: string[];
 
+/**
+ * Specifies a function to dynamically manage the data source when users
+ * change the text in the lookup field. The search function must return
+ * an array or a promise of an array. The `search` property is particularly
+ * useful when the data source does not live in the source code.
+ */
   @Input()
   public search: SkyAutocompleteSearchFunction;
```
Comment (Blackbaud-AlexKingman) [inline src/app/public/modules/lookup/lookup-autocomplete-adapter.ts:55]: The output type `SkyAutocompleteSearchFunction` isn't showing on this docs page, but I'm pretty sure its related to this issue here: https://github.com/blackbaud/skyux-docs-tools/issues/125. Doesn't need to block your PR, just noting it.

File: src/app/public/modules/lookup/lookup-autocomplete-adapter.ts [hunk 4]
```diff
@@ -12,30 +12,79 @@ import {
...
 
+/**
+ * Specifies an array of functions to call against each search result in order
+ * to filter the search results when using the default search function. When
+ * using a custom search function via the `search` property filters must be
+ * applied manually inside that function. The function must return `true` or
+ * `false` for each result to indicate whether to display it in the dropdown list.
+ */
   @Input()
   public searchFilters: SkyAutocompleteSearchFunctionFilter[];
```
Comment (Blackbaud-AlexKingman) [inline src/app/public/modules/lookup/lookup-autocomplete-adapter.ts:82]: This output type `SkyAutocompleteSearchFunctionFilter` is also blocked by the same issue above.

**Review by Blackbaud-AlexKingman — COMMENTED** [review]
Lookup component looks great, just need to slap `@internal` on the `SkyLookupAutocompleteAdapter`.

File: src/app/public/modules/search/search.component.ts [hunk 5]
```diff
@@ -70,31 +70,63 @@ const EXPAND_MODE_NONE: string = 'none';
...
+ * Specifies default search criteria for the input.
+ */
   @Input()
   public searchText: string;
 
+/**
+ * Specifies the expand mode for the search input. The valid options
+ * include `responsive` to collapse the search input into a button on
+ * mobile devices, `none` to *not* collapse the search input on mobile
+ * devices, and fit to extend the search input to `fit` the width of its container.
```
Comment (Blackbaud-AlexKingman) [inline src/app/public/modules/search/search.component.ts:101]: Looks like we need to put the tick marks around the prior `fit` (instead of the latter).

**Review by Blackbaud-AlexKingman — COMMENTED** [review]
Search component looks great. One minor comment and also we'll need to add `@internal` to the `SkySearchAdapterService`.

**Comment by blackbaud-ado** [comment]
Preview build at [https://host.nxt.blackbaud.com/skyux-lookup?_sp.spa=pr-search-lookup-autocomplete-jsdocs](https://host.nxt.blackbaud.com/skyux-lookup?_sp.spa=pr-search-lookup-autocomplete-jsdocs)

**Review by Blackbaud-AlexKingman — CHANGES_REQUESTED** [review]
`SkySearchAdapterService` and `SkyLookupAutocompleteAdapter` will also need to be marked with `@internal`. Then, this is good to go!

**Comment by blackbaud-ado** [comment]
Preview build at [https://host.nxt.blackbaud.com/skyux-lookup?_sp.spa=pr-search-lookup-autocomplete-jsdocs](https://host.nxt.blackbaud.com/skyux-lookup?_sp.spa=pr-search-lookup-autocomplete-jsdocs)

**Review by Blackbaud-AlexKingman — APPROVED** [review]

**Comment by blackbaud-ado** [comment]
Preview build at [https://host.nxt.blackbaud.com/skyux-lookup?_sp.spa=pr-search-lookup-autocomplete-jsdocs](https://host.nxt.blackbaud.com/skyux-lookup?_sp.spa=pr-search-lookup-autocomplete-jsdocs)

**Pull request merged**